src/hooks/useWalletConnect.tsx

```diff
@@ -0,0 +1,94 @@
+import '@walletconnect/react-native-compat'
+
+import { SessionTypes } from '@walletconnect/types'
+import * as React from 'react'
+
+import { walletConnectPromise } from '../components/services/WalletConnectService'
+import { SPECIAL_CURRENCY_INFO } from '../constants/WalletAndCurrencyConstants'
+import { useSelector } from '../types/reactRedux'
+import { WcConnectionInfo } from '../types/types'
+import { getWalletName } from '../util/CurrencyWalletHelpers'
+import { unixToLocaleDateTime } from '../util/utils'
+import { useHandler } from './useHandler'
+import { useWatch } from './useWatch'
+
+interface WalletConnect {
+  getActiveSessions: () => Promise<WcConnectionInfo[]>
+}
+
+/**
+ * Access Wallet Connect
+ */
+export function useWalletConnect(): WalletConnect {
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
+
+  // Utils
+  const publicAddresses = React.useMemo(async () => {
```

Comment:
The `useMemo` hook does not work with async values. If the wallet list updates quickly, you could get a big pile-up of dangling promises.

Instead, you can either use Sam's `useAsyncValue`, which is almost exactly list `useMemo` but handles async calculations, or you can just move this calculation into `getActiveSessions`, which is already `async`, and skip the caching altogether.

Reply:
Yup, just skip the caching and turn this into a method that `getActiveSessions` can call. It's not used anywhere else.